app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -637,6 +640,32 @@ class HomeFragment : Fragment() {
         }.show()
     }
 
+    private fun showRenameTopSitePrompt(topSite: TopSite) {
+        val context = context ?: return
+        val customLayout =
+            LayoutInflater.from(context).inflate(R.layout.top_sites_rename_dialog, null)
+        val topSiteRenameEditText: EditText = customLayout.findViewById(R.id.top_site_name)
+        topSiteRenameEditText.setText(topSite.title)
+
+        AlertDialog.Builder(context).apply {
+            setView(customLayout)
+            setTitle(R.string.top_sites_rename_dialog_title)
+            setNegativeButton(R.string.top_sites_rename_dialog_cancel) { dialog: DialogInterface, _ ->
+                dialog.cancel()
+            }
+            setPositiveButton(R.string.top_sites_rename_dialog_ok) { dialog: DialogInterface, _ ->
+                lifecycleScope.launch(IO) {
+                    with(requireComponents.useCases.topSitesUseCase) {
+                        renameTopSites(topSite, topSiteRenameEditText.text.toString())
```

Comment:
Here I used `.isNotBlank()` on the string to avoid saving a blank title. This is subjective, I guess.